java/src/org/openqa/selenium/grid/config/Config.java

```diff
@@ -59,14 +61,22 @@ default <X> X getClass(String section, String option, Class<X> typeOfClass, Stri
     }
   }
 
+  String DELIM_KEY = "\u001E";
+  String DELIMITER = DELIM_KEY + "=\"record-separator\"";
+
   default List<String> toEntryList(Map<String, Object> mapItem) {
-    return mapItem.entrySet().stream()
-        .map(
-            entry -> {
-              return String.format("%s=%s", entry.getKey(), toJson(entry.getValue()));
-            })
-        .sorted()
-        .collect(ImmutableList.toImmutableList());
+    // transform config settings map into list of key/value strings
+    List<String> entryList = mapItem.entrySet().stream()
+      .map(
+        entry -> {
+          return String.format("%s=%s", entry.getKey(), toJson(entry.getValue()));
+        })
+      .sorted()
```

Comment:
With the addition of an explicit delimiter, sorting is probably no longer necessary.

Reply:
Yes, it would not be needed.

Reply:
Shall I remove the sorting, then?

Reply:
Sure, it can be removed.

Reply:
:heavy_check_mark: I removed the sorting and revised the affected unit tests.